msi/createmsi.py

```diff
@@ -78,78 +35,180 @@ def __init__(self, dirs, files):
         subprocess.check_call(['c:\\Python\python.exe',
                                'c:\\Python\Scripts\\cxfreeze',
                                '--target-dir',
-                               self.staging_dir,
+                               main_stage,
                                '--include-modules',
                                modulestr,
                                'meson.py'])
-        shutil.copy(shutil.which('ninja'), self.staging_dir)
-        if not os.path.exists(os.path.join(self.staging_dir, 'meson.exe')):
+        if not os.path.exists(os.path.join(main_stage, 'meson.exe')):
             sys.exit('Meson exe missing from staging dir.')
-        if not os.path.exists(os.path.join(self.staging_dir, 'ninja.exe')):
+        os.mkdir(ninja_stage)
+        shutil.copy(shutil.which('ninja'), ninja_stage)
+        if not os.path.exists(os.path.join(ninja_stage, 'ninja.exe')):
             sys.exit('Ninja exe missing from staging dir.')
 
-
     def generate_files(self):
-        assert(os.path.isdir(self.staging_dir))
-        comp_ref_xml = ''
-        nodes = {}
-        with open(self.main_xml, 'w') as ofile:
-            for root, dirs, files in os.walk(self.staging_dir):
+        self.root = ET.Element('Wix', {'xmlns': 'http://schemas.microsoft.com/wix/2006/wi'})
+        product = ET.SubElement(self.root, 'Product', {
+            'Name': self.product_name,
+            'Manufacturer': 'The Meson Development Team',
+            'Id': self.guid,
+            'UpgradeCode': self.update_guid,
+            'Language': '1033',
+            'Codepage':  '1252',
+            'Version': self.version,
+            })
+
+        package = ET.SubElement(product, 'Package',  {
+            'Id': '*',
+            'Keywords': 'Installer',
+            'Description': 'Meson %s installer' % self.version,
+            'Comments': 'Meson is a high performance build system',
+            'Manufacturer': 'The Meson Development Team',
+            'InstallerVersion': '200',
+            'Languages': '1033',
+            'Compressed': 'yes',
+            'SummaryCodepage': '1252',
+            })
+
+        if self.bytesize == 64:
+            package.set('Platform', 'x64')
+        ET.SubElement(product, 'Media', {
+            'Id': '1',
+            'Cabinet': 'meson.cab',
+            'EmbedCab': 'yes',
+            })
+        targetdir = ET.SubElement(product, 'Directory', {
+            'Id': 'TARGETDIR',
+            'Name': 'SourceDir',
+            })
+        progfiledir = ET.SubElement(targetdir, 'Directory', {
+            'Id' : self.progfile_dir,
+            })
+        installdir = ET.SubElement(progfiledir, 'Directory', {
+            'Id': 'INSTALLDIR',
+            'Name': 'Meson'})
+
+        ET.SubElement(product, 'Property', {
+            'Id': 'WIXUI_INSTALLDIR',
+            'Value': 'INSTALLDIR',
+            })
+        ET.SubElement(product, 'UIRef', {
+            'Id': 'WixUI_FeatureTree',
+            })
```

Comment:
__[flake8]__
* [E123] closing bracket does not match indentation of opening bracket's line

<!-- Comment by [SideCI](https://sideci.com) -->